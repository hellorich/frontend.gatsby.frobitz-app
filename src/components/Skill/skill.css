.skill {
  color: #FFF;
  display: inline-block;
  margin: 0.25rem;
  min-width: 5rem;
  padding: 0.5rem;
  text-align: center;
  text-decoration: none;
}

.skill--advanced {
  background-color: rgb(33, 150, 243);
  
  &:active, &:focus, &:hover, &:link, &:visited {
    background-color: rgb(33, 150, 243);
  }
}

.skill--basic {
  background-color: rgb(33, 201, 243);
  
  &:active, &:focus, &:hover, &:link, &:visited {
    background-color: rgb(33, 201, 243);
  }
}

.skill-list--item {
  list-style: none;
}