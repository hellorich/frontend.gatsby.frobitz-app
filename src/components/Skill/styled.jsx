import styled from "styled-components"

export const StyledSkill = styled.li`
  list-style: none;

  .skill {
    color: #FFF;
    display: inline-block;
    margin: 0.25rem;
    min-width: 5rem;
    padding: 0.5rem;
    text-align: center;
    text-decoration: none;
    
    &.skill--advanced {
      background-color: rgb(33, 150, 243);
      
      &:active, &:focus, &:hover, &:link, &:visited {
        background-color: rgb(33, 150, 243);
      }
    }
    
    &.skill--basic {
      background-color: rgb(33, 201, 243);
      
      &:active, &:focus, &:hover, &:link, &:visited {
        background-color: rgb(33, 201, 243);
      }
    }
  }
`