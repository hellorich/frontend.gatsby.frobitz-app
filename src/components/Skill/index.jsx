import React from 'react'
import PropTypes from 'prop-types'
import './skill.css'

export default function Skill ({ skill: { title, level, link } }) {
  const skillLevel = `skill--${level}`
  return (
    <li className="skill-list--item">
      <a 
        className={['skill', skillLevel].join(' ')}
        href="{link}"
        rel="external"
      >
      {title}
      </a>
    </li>
  )
}

Skill.propTypes = {
  skill: PropTypes.shape({
    title: PropTypes.string.isRequired,
    link: PropTypes.string.isRequired,
    level: PropTypes.oneOf(['basic', 'advanced']),
  })
}