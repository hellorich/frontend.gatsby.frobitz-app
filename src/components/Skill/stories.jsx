import React from 'react'
import { Skill } from 'components/Skill'

export default {
  component: Skill,
  title: 'Skill',
}

const Template = args => <Skill {...args} />

export const Default = Template.bind({})
Default.args = {
  skill: {
    title: 'Test Skill',
    level: 'advanced',
    link: 'https://developer.mozilla.org',
  },
}

export const Basic = Template.bind({})
Basic.args = {
  skill: {
    ...Default.args.skill,
    level: 'basic',
  },
}