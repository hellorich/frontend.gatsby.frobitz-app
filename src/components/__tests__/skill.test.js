import React from 'react'
import renderer from 'react-test-renderer'
import Skill from '../Skill/Skill'

describe('Skill', () => {
  it('renders correctly', () => {
    const args = {
      skill: {
        title: 'Test Skill',
        level: 'advanced',
        link: 'https://developer.mozilla.org'
      },
    }
    
    const component = renderer.create(
      <Skill {...args} />
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})