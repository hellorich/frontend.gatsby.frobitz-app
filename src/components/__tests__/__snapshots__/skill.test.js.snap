// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skill renders correctly 1`] = `
<li
  className="skill-list-item"
>
  <a
    className="skill skill--advanced"
    href="{link}"
    rel="external"
  >
    Test Skill
  </a>
</li>
`;
